File: javascript-logger-webjar/src/main/kotlin/com/github/forinil/javascriptlogger/webjar/Logger.kt
package com.github.forinil.javascriptlogger.webjar

import com.github.forinil.javascriptlogger.webjar.util.LogData
import org.w3c.xhr.XMLHttpRequest

/**
* Created by Konrad Botor on 15.04.2017T15:40
* in package com.github.forinil.javascriptlogger.webjar in project JavaScriptLogger.
*/
object Logger {
    @JsName("info")
    fun info(page: String, function: String, message: String) {
        console.info("[INFO] $page::$function - $message")
        val logData = LogData("INFO", page, function, message)
        sendPostRequest(logData)
    }

	@JsName("debug")
    fun debug(page: String, function: String, message: String) {
        console.info("[DEBUG] $page::$function - $message")
        val logData = LogData("DEBUG", page, function, message)
        sendPostRequest(logData)
    }

	@JsName("error")
    fun error(page: String, function: String, message: String, errorCode: Int) {
        console.error("[ERROR] $page::$function - $message - $errorCode")
        val logData = LogData("ERROR", page, function, message, errorCode)
        sendPostRequest(logData)
    }

	@JsName("warn")
    fun warn(page: String, function: String, message: String) {
        console.warn("[WARN] $page::$function - $message")
        val logData = LogData("WARN", page, function, message)
        sendPostRequest(logData)
    }

	@JsName("trace")
    fun trace(page: String, function: String, message: String) {
        console.info("[TRACE] $page::$function - $message")
        val logData = LogData("TRACE", page, function, message)
        sendPostRequest(logData)
    }

    private fun sendPostRequest(logData: LogData) {
        val xhttp = XMLHttpRequest()
        xhttp.open("POST", "logger", true)
        xhttp.send(logData.toJSON())
    }
}

File: javascript-logger-testapp/src/main/webapp/static/js/logger.js
var logger = this["javascript-logger-webjar"].com.github.forinil.javascriptlogger.webjar.Logger

function info() {
	var message = document.getElementById('messageText').value;
	logger.info('index', 'info_test', message)
}

function debug() {
	var message = document.getElementById('messageText').value;
	logger.debug('index', 'debug_test', message)
}

function warn() {
	var message = document.getElementById('messageText').value;
	logger.warn('index', 'warn_test', message)
}

function trace() {
	var message = document.getElementById('messageText').value;
	logger.trace('index', 'trace_test', message)
}

function error() {
	var message = document.getElementById('messageText').value;
	var errorCode = document.getElementById('errorCode').value;
	if (errorCode === '') {
		errorCode = null
	}
	logger.error('index', 'error_test', message, errorCode)
}

window.onerror = function (msg, url, lineNo, columnNo, error) {
    var string = msg.toLowerCase();
    var substring = "script error";
    if (string.indexOf(substring) > -1){
        logger.error('index', 'onError', 'Script Error: See Browser Console for Detail');
    } else {
        var message = [
            'Message: ' + msg,
            'URL: ' + url,
            'Line: ' + lineNo,
            'Column: ' + columnNo,
            'Error object: ' + JSON.stringify(error)
        ].join(' - ');

        logger.error('index', 'onError', message, -1);
    }

    return false;
};